src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -435,14 +435,23 @@ private function parseController(array|object|string|null $controller): array|st
 
         if (\is_array($controller)) {
             try {
+                $rc = new \ReflectionClass($controller[0]);
                 $r = new \ReflectionMethod($controller[0], $controller[1]);
 
-                return [
+                $data = [
                     'class' => \is_object($controller[0]) ? get_debug_type($controller[0]) : $controller[0],
                     'method' => $controller[1],
                     'file' => $r->getFileName(),
                     'line' => $r->getStartLine(),
                 ];
+
+                if ($r->getFileName() !== $rc->getFileName()) {
+                    $data['methodClass'] = $r->class;
```

Comment:
Does it work with traits?

Reply:
`if ($r->class !== $rc->name || $r->getFileName() !== $rc->getFileName()) {` maybe?

Reply:
@nicolas-grekas `$r->class` or `$rc->name` will always reference the inheriting class and not the trait. The path to the method is already correct because `$r->getFileName()` references the trait in line 444

Reply:
@nicolas-grekas makes sense. i got confused by my own goals -_-. Changed the line and added a test for traits. Thank you.